src/main/java/org/folio/edge/sip2/handlers/freemarker/FreemarkerRepository.java

```diff
@@ -0,0 +1,79 @@
+package org.folio.edge.sip2.handlers.freemarker;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateExceptionHandler;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.edge.sip2.parser.Command;
+
+public class FreemarkerRepository {
+
+  private static FreemarkerRepository instance;
+  private HashMap<Command, Template> templates;
+  private final Logger log;
+
+  /**
+   * Static method to get the only running Freemarker Repository instance.
+   */
+  public static FreemarkerRepository getInstance() {
+    if (instance == null) {
+      synchronized (FreemarkerRepository.class) {
+        if (instance == null) {
+          instance = new FreemarkerRepository();
+        }
+      }
+    }
+    return instance;
+  }
+
+  /**
+   * Get the template by Command.
+   *
+   * @param command the Command that serves as a Key to the runtime cache
+   * @return Compiled freemarker template
+   */
+  public Template getFreemarkerTemplate(Command command) {
+    return templates.get(command);
+  }
+
+  private FreemarkerRepository() {
+    initializeTemplates();
+    log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
Good catch about the NPE. As for creating the logger in the constructor, I think that's the right place for it, unless you think there's a better place, maybe inline at variable declaration?

Reply:
Typically, the pattern is to declare it as a private static final as in:
`private static final Logger log = LogManager.getLogger();`

Since these are one instance classes, I am fine with them being non-static, but I'd declare them at object creation like:
`private final Logger log = LogManager.getLogger();`

I've not seen a logger created in the constructor before, since logging is typically outside the scope of the business logic of the class. Some frameworks inject loggers so you don't have to worry about it.

Reply:
Yeah, i think part of my motivation for instantiating it in the constructor is to be able to inject it in for testing purposes, but I thought that we could mock it via Mockito somehow. I briefly looked into this over the weekend but not able to get Mockito to mock it.  And I didn't want to go overboard with the testing for now, so we can revisit it later.

Reply:
You want to mock the logger?

Reply:
if you want 100% coverage or to verify with certainty every line of code! :)  I don't think I want to do that now. given that some negative test cases already run by these logging statements without verifying them, and that we have bigger fish to fry.

Reply:
But we don't make decisions based on what we log, do we? As long as the test gets into the block with the log line, the line is covered, at least that is my understanding. I assume something else more tangible will happen in addition to logging, like throwing an exception or returning `null` that can be checked in the test.

Reply:
We don't, but sometimes an error message is critical for troubleshooting, for ease of fishing it out of the logs, so we'd like to make sure that it's there.

Reply:
Sure, but I am not convinced we need to do things that are contrary to best practices in order to ensure messages, that could be suppressed/disable via configuration, are exactly what we expect. I think we should use something built into log4j2, like a plugin: https://stackoverflow.com/questions/24205093/how-to-create-a-custom-appender-in-log4j2

This way, we add it for the test, get the log4j2 messages in the appender plugin and assert they are what we expect.

Here is how someone did that for log4j1.x: https://stackoverflow.com/questions/1827677/how-to-do-a-junit-assert-on-a-message-in-a-logger